products.js

```diff
@@ -227,15 +228,15 @@ class DiffStats {
         if (this.feature === 'image') {
           //compare images by src
           expectedText = expectedDom.body.firstChild.src;
-          gotText = this.tuningRoutine(sourceDom, dict).map(fnode => fnode.element.src)[0];
+          gotText = this.tuningRoutine()(sourceDom).map(fnode => fnode.element.src)[0];
```

Comment:
Here's one last occurrence of this dispatch. Maybe we can move it into the `tunedWhatsits` variable where you parked the other pieces. It may be natural to roll all the things that differ based on the dispatch key into classes and have your mapping point to those as values.